packages/common/src/controllers/CheckoutController.ts

```diff
@@ -70,23 +72,31 @@ export default class CheckoutController {
   };
 
   updateOrder = async (order: Order, paymentMethodId?: number, couponCode?: string | null): Promise<void> => {
-    const updateOrderPayload: UpdateOrderPayload = {
-      order,
-      paymentMethodId,
-      couponCode,
-    };
+    let response;
+
+    try {
+      response = await this.checkoutService.updateOrder({ order, paymentMethodId, couponCode });
+    } catch (error: unknown) {
+      // TODO: we currently (falsely) assume that the only error caught is because the coupon is not valid, but there
+      //  could be a network failure as well (JWPCheckoutService)
+      throw new FormValidationError({ couponCode: [i18next.t('account:checkout.coupon_not_valid')] });
```

Comment:
Should it be a form error in this case?

Reply:
Although changed, it works exactly like before. The only error thrown by the JWP service is `Invalid coupon code`. We will have to improve that first before we can throw a more specific form error here.